scripts.Rmd

```diff
@@ -0,0 +1,258 @@
+
+```{r}
+
+simple_config <- drake_config(simple_plan)
+vis_drake_graph(simple_config)
+
+```
+
+
+## Building the connections
+
+Just as the scripts needed to run in a certain order, so do our targets now. We
+pass targets as function arguments to express this execution order.
+
+For example, when we write `munged_data = munge_data(data)`, we are signaling to
+`drake` that the `munged_data` target depends on the function `munge_data()` and 
+the target data.
+
+We can apply the same logic by passing the targets of the `code_to_function()`
+generated functions as arguments in the dependent script-functions.
+
+```{r}
+
+script_based_plan <- drake_plan(
+  data        = load_data(),
+  munged_data = munge_data(data),
+  hist        = make_histogram(munged_data),
+  fit         = do_regression(munged_data),
+  report      = generate_report(hist, fit)
+)
+```
+
+```{r echo=FALSE}
+
+script_based_config <- drake_config(script_based_plan)
+vis_drake_graph(script_based_config)
+
+```
+
+
+## Keeping up to date
+
+Any time we change a script, we need to run `code_to_function()` again to make 
+sure we have an up-to-date script-function. `drake` can see that the function is 
+updated and when `make()` is called, it will rerun the updated function and all
+down-stream functions that rely on the output. 
+
+For example, let's fine tune our histogram. We open 03_histogram.R, change the 
+binwidth argument from 0.5 to 0.25, and call code_to_function("03_histogram.R")
+all over again.
+
+```{r echo=FALSE}
+writeLines(
+  c("munged_data <- readRDS(\"data/munged_data.RDS\")",
+    "gg <- munged_data %>%",
+    "ggplot() +",
+    "geom_histogram(",
+    "  aes(",
+    "    x = Petal.Width,",
+    "    fill = Species",
+    "    ),",
+    "  binwidth = 0.5) +",
+    "  theme_gray(20)",
+    "ggsave(",
+    "filename = \"data/Petal_Width_vs_Species.PNG\",",
+    "plot = gg",
+    ")",
+    "saveRDS(gg, \"data/Petal_Width_vs_Species.RDS\")"),
+  "03_histogram.R"
+  )
+
+```
+
+```{r}
+# We need to rerun code_to_function() to tell drake that the script changed.
+make_histogram <- code_to_function("03_histogram.R")
+```
+
+Targets `hist` and `report` depend on the code we modified, so `drake` marks 
+those targets as outdated.
+
+```{r echo=FALSE,message=FALSE}
+outdated_targets<-outdated(script_based_config)
+#needs to be run, for some reason `drake` is not recognizing the update when compiling
+vis_drake_graph(script_based_config, targets_only = TRUE)
+```
+
+When you call `make()`, `drake` runs `make_histogram()` because the underlying 
+cript changed, and it runs `generate_report()` because the report depends on 
+`hist`.
+
+```{r}
+make(script_based_plan)
+```
+
+The dependency graph is now all up to date!
+
+```{r, echo=FALSE}
+vis_drake_graph(script_based_config, targets_only = TRUE)
+```
+
+
+## `drake` and scripts
+
+> A large number of preexisting data science workflows consist of numbered,
```

Comment:
I am not sure you need to quote yourself here.

Reply:
At the beginning of the next paragraph, maybe just say "`code_to_function()` lets `drake` to accommodate script-based projects too big and cumbersome to refactor."